### PR TITLE
#1328 Correct c++11 compilation with clang

### LINKED ISSUE #1328: Clang compilation is broken

### PR DESCRIPTION
**Pull request by lecourtoisflo**
closes #1328

### CODE REVIEW

File: dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ComposableDocumentHandler.h [hunk 1]
```diff
@@ -38,7 +38,15 @@ namespace parser {
  */
 class ComposableDocumentHandler : public DocumentHandler, private ComposableBase {
 public:
+  /**
+   * @brief Destructor
+   */
+#ifdef LANG_CXX11
+  // not using override keyword generates a warning, processed as an error, in dynawo
+  ~ComposableDocumentHandler() override {}
```
Comment (rosiereflo) [inline dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ComposableDocumentHandler.h:46]: no need of a virtual?

**Review by rosiereflo — CHANGES_REQUESTED** [review]
libxml used to compile, is this triggered by more recent versions of clang?

File: dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ComposableDocumentHandler.h (hunk 1 above)
Comment (lecourtoisflo) [reply dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ComposableDocumentHandler.h:46]: override implies virtual

Comment (lecourtoisflo) [reply dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ComposableDocumentHandler.h:46]: override implies virtual

**Comment by lecourtoisflo** [comment]
> libxml used to compile, is this triggered by more recent versions of clang?

I don't know. @gautierbureau discovered that yesterday. The problem is seen only in C++11

**Comment by gautierbureau** [comment]
I don't know what causes the issue and why it was not seen before... It seems we get the warnings from libxml and maybe some SYSTEM keywork should be added to target_include_directories when we add the headers of libxml

**Review by rosiereflo — APPROVED** [review]

**Review by gautierbureau — APPROVED** [review]

**Pull request merged**